src/cytoscape-node-html-label.ts

```diff
@@ -115,10 +115,18 @@ interface CytoscapeContainerParams {
     }
 
     updateData(data: any) {
-      try {
-        this._node.innerHTML = this.tpl(data);
-      } catch (err) {
-        console.error(err);
+      while (this._node.firstChild) {
+        this._node.removeChild(this._node.firstChild);
+      }
+      const children = new DOMParser()
+                           .parseFromString(this.tpl(data), 'text/html')
+                           .body.children;
+      for (const el of Array.from(children)) {
+        try {
+          this._node.appendChild(el);
+        } catch (err) {
+          console.error(err);
```

Comment:
Could you add more detail on why this error happened? e.g. `Error parsing node html label:`

Reply:
We are working on a high security system that that does not permit assigning to innetHTML for security reasons.

our linter when building our project for production is showing us error: Assigning directly to Element#innerHTML can result in XSS vulnerabilities. If the assigned value does not contain HTML, assign it to the DOM element's textContent property.

Reply:
Has the core vuln been addressed?  ie, can we still add "<img src='x' onerror='alert(1)'>"; as a value?   generally it's good practice not to make changes purely to silence static analysis without addressing actual security issue.

Reply:
> Has the core vuln been addressed?

I think it only silences the vulnerability issue on his linter, which IIUC won't let him have this. The "vulnerability" is still there AFAIK. The only improvement I can see is that we are throwing an error on invalid html.

Perhaps we should also warn about this on the docs. Something like "the template is parsed like this ... and you are still in charge of providing safe HTML" WDYT @blazespinnaker ?